### PR TITLE
Resolve an issue with CASSANDRA_HOME.

### PR DESCRIPTION
**Pull request by tylerhunt**
The Cassandra RubyGem and Cassandra itself have different ideas about what `CASSANDRA_HOME` should point to, so if a custom `CASSANDRA_INCLUDE` is specified, assume that it will define `CASSANDRA_HOME` and `CASSANDRA_CONF`.

The gist of this fix is that if you're trying to use a location for Cassandra other than the default `~/cassandra`, the environment variables quickly get you into trouble, since the Gem expects `CASSANDRA_HOME` to point to the directory above the actually Cassandra root (e.g. `~/cassandra`), and Cassandra expects it to point to the root (e.g. `~/cassandra/cassandra-0.8`).

The intention of this fix is to assume that if you're using a custom `CASSANRA_INCLUDE` file, that you know that you need to define `CASSANDRA_HOME` and `CASSANDRA_CONF` yourself. This allows you to have a `CASSANDRA_HOME` defined for `cassandra_helper` to use, while still allowing that value to be overridden by the `CASSANRA_INCLUDE` file.

There may well be a better way to resolve this, but this seem like it might cause the least friction. I'm surprised others haven't run into this before, which very well may be a good indication that I'm fundamentally confused about some aspect of how these pieces are supposed to fit together.

In effect, though, what I believe should happen is that the `Rakefile` should give the include file a chance to override the currently set environment variables. Alternatively, if using a `cassandra.in.sh` file other than the one that ships with Cassandra, removing the conditional checks for the presence of those environment variables would work, too, and this may be preferable to making the change here.

Either way, it seems to me that the issue of the discrepancy between what the Gem is expecting `CASSANDRA_HOME` to point to and what Cassandra is expecting should be addressed.

Sorry for the rambling, but my hope is that this request will open up a discussion about what I see as a potential point of confusion.

### CODE REVIEW

**Comment by rwjblue** [comment]
I am not quite sure that I understand the scenario that you are describing.  Could you describe what you are trying to do?

The cassandra_helper script simply provides a simple way to get up and running quickly.  I wouldn't imagine anyone would want to use these rake functions to manage a live production cluster.

I certainly do not have an issue making the gem use CASSANDRA_HOME in the same way as Cassandra itself.  It seems to me that in almost all cases we ultimately want the actual Cassandra location anyways...

**Comment by tylerhunt** [comment]
Okay, maybe I'm just misunderstanding what `cassandra_helper`'s purpose is, then.

My goal was to have a directory structure that looked like `#{Rails.root}/db/cassandra/cassandra-0.8`, with the configuration under `#{Rails.root}/config/cassandra`. In this case, I'd set the `CASSANDRA_INCLUDE` to point to `#{Rails.root}/config/cassandra/cassandra.in.sh` and have that script set up `CASSANDRA_HOME` to point to `#{Rails.root}/db/cassandra/cassandra-0.8` and `CASSANDRA_CONF` to point to `#{Rails.root}/config/cassandra`.

The problem, though, is that `cassandra_helper` (i.e. the `Rakefile`) relies on `CASSANDRA_HOME` to point to `#{Rails.root}/db/cassandra`, which means I have to use a custom `CASSANDRA_HOME` that does something like this:

```
CASSANDRA_CONF=`dirname "${BASH_SOURCE[0]}"`
CASSANDRA_HOME="${CASSANDRA_CONF}/../../db/cassandra/cassandra-0.8"
```

Instead of something more like the default include, which will use the existing environment variables if they're already set:

```
if [ "x$CASSANDRA_HOME" = "x" ]; then
    CASSANDRA_HOME=`dirname $0`/..
fi

if [ "x$CASSANDRA_CONF" = "x" ]; then
    CASSANDRA_CONF=$CASSANDRA_HOME/conf
fi
```

Basically, if think if I could set `CASSANDRA_HOME` to the Cassandra's definition of it then I wouldn't have to resort to using a custom `CASSANDRA_INCLUDE`.
